src/Http/Auth/RequestAuthenticatorMiddleware.php

```diff
@@ -81,18 +84,24 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
      */
     public function authenticate(ServerRequestInterface $request): void
     {
-        // For requests to public routes, that provide extra information to Authenticated
-        // Users. eg. show contributors
-        try {
-            $this->authenticateToken($request);
-        } catch (\Exception $exception) {
-            if ($this->isCorsPreflightRequest($request) || $this->isPublicRoute($request)) {
-                $this->token = null;
-                return;
+        if ($this->isCorsPreflightRequest($request)) {
+            return;
+        }
+
+        if ($this->isPublicRoute($request)) {
+            if ($this->authenticatePublicRoutes) {
+                try {
+                    $this->authenticateToken($request);
+                } catch (\Exception $exception) {
+                    $this->token = null;
+                    return;
+                }
```

Comment:
```php
    $isPublic = $this->isPublicRoute($request);

    if ($isPublic && $this->authenticatePublicRoutes) {
      try {
        $this->authenticateToken($request);
      } catch (\Exception $exception) {
        $this->token = null;
      }
    }

    if ($isPublic) {
      return;
    }
```

I think you could write it less nested like this.

Also you could rename `authenticatePublicRoutes` to `shouldAuthenticatePublicRoutes`, down to preferences.

Reply:
Or even with an extra early return statement:

```
if ($isPublic && !$this->authenticatePublicRoutes) {
  return;
}

if ($isPublic) {
  try {
    $this->authenticateToken($request);
  } catch (\Exception $exception) {
    $this->token = null;
  }
}
```

Reply:
Yeah, but If I'm not mistaken you also have to return in both cases:

When authenticatePublicRoutes is true or false.
So you would have to ass a return statement after the try catch

Reply:
Indeed, you have to return in both cases regardless of `authenticatePublicRoutes`.